smdebug/tensorflow/base_hook.py

```diff
@@ -489,12 +491,14 @@ def set_optimizer_variables(self, optimizer_variables):
         # From TF 2.2, executing_eagerly_outside_functions() can be used as
         # ops.executing_eagerly_outside_functions() or tf.compat.v1.executing_eagerly_outside_functions().
         # But in TF 2.1, only ops.executing_eagerly_outside_functions() is valid
-        if is_tf_version_2x() and ops.executing_eagerly_outside_functions():
```

Comment:
why is this removed?

Reply:
this check was added to block set_optimizer_variables in eager mode until this PR which ensures capturing optimizer variable is possible